packages/mui-joy/src/styles/extendTheme.ts

```diff
@@ -644,5 +645,38 @@ export default function extendTheme(themeOptions?: CssVarsThemeOptions): Theme {
     });
   };
 
+  // @ts-ignore palette could be defined in the ThemeProvider's theme
+  theme.palette = deepmerge(themeOptions?.palette || {}, {
+    ...(theme.colorSchemes.light.palette as RuntimeColorSystem['palette']),
+    colorScheme: 'light',
+  });
```

Comment:
I wonder if `extendTheme()` should not resolve the palette yet.

My initial intention is that `extendTheme()` should return sets of colors for your application but it should not decide what the current color scheme is. That logic will be handled by `CssVarsProvider`.